ui_tests/pages/inventory_page.py

```diff
@@ -0,0 +1,336 @@
+        )
+        sec_engine = driver.find_element(By.XPATH, sec_engine_field_xpath)
+        helper.clear_input(sec_engine)
+        sec_engine.send_keys(walk_interval)
+        time.sleep(1)
+
+    def set_smart_profiles(self, param):
+        logger.info(f"set inventory device smart profiles enabled to: {param}")
+        if param == "true" or param == "false":
+            smart_profile_true_xpath = (
+                f"//button[@data-test='sc4snmp:form:smart-profile-{param}']"
+            )
+            option = driver.find_element(By.XPATH, smart_profile_true_xpath)
+            option.click()
+        else:
+            logger.error(
+                f"Wrong parameter specified. Expected: true or false, received: {param}"
+            )
+
+    def select_profiles(self, profiles, edit=False):
+        logger.info(f"select profiles: {profiles}")
+        profiles_input_xpath = (
+            "//div[@data-test='sc4snmp:form:profiles-multiselect']//div//input"
+        )
+        profile_input = driver.find_element(By.XPATH, profiles_input_xpath)
+        if edit:
+            profile_options_xpath = "//button[@data-test='selected-option']"
+            options = driver.find_elements(By.XPATH, profile_options_xpath)
+            for option in options:
+                option.click()
+                time.sleep(0.5)
+        time.sleep(1)
+        for profile in profiles:
+            profile_input.send_keys(profile)
+            profile_input.send_keys(Keys.ENTER)
+            time.sleep(2)
+            # we need to hide profile list,
+            # otherwise it can break test execution and popup can intercept clicking on smart profiles
+            profile_input.send_keys(Keys.ESCAPE)
+
+    def _get_inventory_data(self, host, field):
+        if field == "snmp_version":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-version' and ancestor::tr//td[text()='{host}']]"
+        elif field == "port":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-port' and ancestor::tr//td[text()='{host}']]"
+        elif field == "community_string":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-community' and ancestor::tr//td[text()='{host}']]"
+        elif field == "secret":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-secret' and ancestor::tr//td[text()='{host}']]"
+        elif field == "security_engine":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-security-engine' and ancestor::tr//td[text()='{host}']]"
+        elif field == "walk_interval":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-walk-interval' and ancestor::tr//td[text()='{host}']]"
+        elif field == "profiles":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-profiles' and ancestor::tr//td[text()='{host}']]"
+        elif field == "smart_profiles":
+            field_xpath = f"//td[@data-test='sc4snmp:inventory-smart-profiles' and ancestor::tr//td[text()='{host}']]"
```

Comment:
wouldn't it be better to keep it in dictionary?

Reply:
yes, I was trying to use switch case, but dict is better :)